### PR TITLE
Fix matchAny and matchFeatures processing logic for image compatibility

### PR DESCRIPTION
**Pull request by mfranczy**
If matchAny and matchFeatures are defined at the same level in the spec, you might receive a false positive.

### CODE REVIEW

**Comment by mfranczy** [comment]
Thanks @ArangoGutierrez, during additional tests I discovered that we don't handle matchAny and matchFeatures correctly when these are together. I changed the PR to bugfix.

**Comment by mfranczy** [comment]
/hold cancel

All good.

**Comment by mfranczy** [comment]
> Thanks for the PR. The changes look good overall, but I noticed we could improve test coverage for some edge cases:
> 
>     * We should add a test case for when the input array is empty
> 
>     * (Optional) It would be good to test the scenario where `MatchFeatures` is false but `MatchAny` is true

@Xunli-Yang fixed. Added tests for empty spec and extended tests to simulate the optional behaviour you mentioned.

File: pkg/apis/nfd/nodefeaturerule/rule.go [hunk 1]
```diff
@@ -133,7 +134,15 @@ func Execute(r *nfdv1alpha1.Rule, features *nfdv1alpha1.Features, failFast bool)
...
-	matchStatus.IsMatch = isMatch
+
+	switch {
+	case len(r.MatchFeatures) > 0 && len(r.MatchAny) > 0:
+		matchStatus.IsMatch = isMatchAny && isMatchFeature
+	case len(r.MatchFeatures) > 0:
+		matchStatus.IsMatch = isMatchFeature
+	case len(r.MatchAny) > 0:
+		matchStatus.IsMatch = isMatchAny
+	}
```
[the remark this answers is not in the document]
Comment (mfranczy) [reply pkg/apis/nfd/nodefeaturerule/rule.go]: That actually is a nice suggestion :)

**Comment by Xunli-Yang** [comment]
> > Thanks for the PR. The changes look good overall, but I noticed we could improve test coverage for some edge cases:
> > ```
> > * We should add a test case for when the input array is empty
> > 
> > * (Optional) It would be good to test the scenario where `MatchFeatures` is false but `MatchAny` is true
> > ```
> 
> @Xunli-Yang fixed. Added tests for empty spec and extended tests to simulate the optional behaviour you mentioned.
Thanks @mfranczy , it looks good to me.

**Comment by ChaoyiHuang** [comment]
/LGTM thanks for the cases to enhance test coverage
